src/states/MovingState.java

```diff
@@ -0,0 +1,32 @@
+package states;
+
+import elevator.Elevator;
+import events.ArrivalEvent;
+import events.FloorEvent;
+
+public class MovingState extends State {
+
+	public MovingState(Elevator e) {
+		super(e);
+	}
+	
+	@Override
+	public State handleFloorEvent(FloorEvent e) {
+		if (e.getDestination() == elevator.getCurrentFloor()) {
+			// this part might need to fix. Need to set elevators state to stationary then make elevator call handle floor event again
+			// elevator.handleFloorEvent(e);

```

Comment:
@krishane2018 Do you think it's better if the elevator calls handleFloorEvent again or should the state take care of all of that? (If the state should, I might need to revert back to my old design)